src/System.Security.Cryptography.X509Certificates/tests/X509StoreTests.cs

```diff
@@ -539,6 +539,11 @@ public static void UnixCannotModifyDisallowedStore(bool useEnum, OpenFlags openF
         [PlatformSpecific(TestPlatforms.Linux)] // Windows/OSX doesn't use SSL_CERT_{DIR,FILE}.
         private void X509Store_MachineStoreLoadSkipsInvalidFiles()
         {
+            if (geteuid() == 0)
+            {
+                throw new SkipTestException("Test can't be run as root"); // root can read '2.pem'
```

Comment:
You need to change the [Fact] to be [ConditionalFact] for SkipTestException to have the desired effect.

Reply:
Though I wonder if we should instead have:
```C#
public static bool NotRunningAsRoot => geteuid() != 0;
```
and then do:
```C#
[ConditionalFact(nameof(NotRunningAsRoot))]
```
to be more declarative about it.

Reply:
Yes please only use SkipTestException if it's absolutely necessary. A conditional fact should probably suffice here.

Reply:
Fixed!